internal/controllers/watchdog/controller.go

```diff
@@ -0,0 +1,71 @@
+package watchdog
+
+import (
+	"context"
+	"time"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/Azure/eno/internal/manager"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// watchdogController exposes metrics that track the states of Eno resources relative to the current time.
+// The idea is to identify deadlock states so they can be alerted on.
+type watchdogController struct {
+	client    client.Client
+	threshold time.Duration
+}
+
+func NewController(mgr ctrl.Manager, threshold time.Duration) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Composition{}).
+		WithLogConstructor(manager.NewLogConstructor(mgr, "watchdogController")).
+		Complete(&watchdogController{
+			client:    mgr.GetClient(),
+			threshold: threshold,
+		})
+}
+
+func (c *watchdogController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	list := &apiv1.CompositionList{}
+	err := c.client.List(ctx, list)
```

Comment:
I assume that it's safe to list them all because this is coming from the informer, but just want to sanity check.